### ISSUE TITLE
Package request: photoview

### ISSUE THREAD

**Issue by Skallwar** [NixOS/nixpkgs#217236]
**Project description**

Photoview is a simple and user-friendly photo gallery that's made for photographers and aims to provide an easy and fast way to navigate directories, with thousands of high-resolution photos.

**Metadata**

* homepage URL: https://photoview.github.io/
* source URL: https://github.com/photoview/photoview
* license: agpl3
* platforms: linux

**Comment by Skallwar** [comment]
I tried to packaged it myself here: https://gist.github.com/Skallwar/e68c10b4380e6d01a99636d3cc9e55ac

The go part is working but I had some trouble for the `ui` part. I tried using node2nix and it work with this patch: https://gist.github.com/Skallwar/a3c0013d6fd89b62f0951dccf7fca43d
But I failed to integrate this to nixpkgs because I didn't understood how to apply the patch.

**Comment by pinn3** [comment]
It looks like you use `buildNpmPackage` and not `node2nix`. It accepts a `patches` list so you should just be able to provide it there:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/node/build-npm-package/default.nix#L8

**Comment by Skallwar** [comment]
Yes on the gist I try to used buildNpmPackage but it doesn't work I got some `ENOTCACHED` error. That's why there are some fetchurl commented. My technique worked for the first dependency but it hanged for the second one.

I tried to use node2nix afterwards which solved the `ENITCACHED` issue but I need to apply the patch and specify the source and I don't know how to do that on the generated files.

I'm still new to nix so I hope what I wrote makes sense :)

**Comment by dit7ya** [comment]
@Skallwar The current `buildNpmPackage` implementation has a bug - that's the reason for the `ENOTCACHED` error. 

@winterqt has a WIP fix for that.

**Comment by aualbert** [comment]
@Skallwar Has [this commit](https://github.com/NixOS/nixpkgs/pull/214454) fixed the bug?
